File: apps/twitch/src/triggers/come.ts
import type { Bot } from '@/bot/bot';
import type { Trigger } from '@/triggers/trigger';
import type { EventSubChannelChatMessageEvent } from '@twurple/eventsub-base';

export class ComeTrigger implements Trigger {
  triggerChance = 20;

  shouldTrigger(event: EventSubChannelChatMessageEvent): boolean {
    const triggers = ['come', 'coming', 'cum', 'came'];
    const words = event.messageText.toLowerCase().split(/\W+/).filter(Boolean)
    return words.some((word) => triggers.includes(word))
  }

  async execute(bot: Bot, event: EventSubChannelChatMessageEvent): Promise<void> {
    await bot.sendMessage('no coming', event.messageId);
  }
}

File: twitch/internal/triggers/come.go
package triggers

import (
	"slices"
	"strings"

	"github.com/joeyak/go-twitch-eventsub/v3"
	"github.com/lukeramljak/charsibot/internal/bot"
)

type ComeTrigger struct {
	triggers []string
}

func NewComeTrigger() *ComeTrigger {
	return &ComeTrigger{
		triggers: []string{"come", "coming", "cum", "came"},
	}
}

func (t *ComeTrigger) TriggerChance() int {
	return 20
}

func (t *ComeTrigger) ShouldTrigger(event twitch.EventChannelChatMessage) bool {
	// Split by non-word characters and filter empty strings
	words := strings.FieldsFunc(strings.ToLower(event.Message.Text), func(r rune) bool {
		return !((r >= 'a' && r <= 'z') || (r >= '0' && r <= '9'))
	})

	for _, word := range words {
		if slices.Contains(t.triggers, word) {
			return true
		}
	}
	return false
}

func (t *ComeTrigger) Execute(b *bot.Bot, event twitch.EventChannelChatMessage) {
	b.SendMessage(bot.SendMessageParams{
		Message:              "no coming",
		ReplyParentMessageID: event.MessageId,
	})
}
